hooks/pre-commit

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+RED_BG="\033[41m"
+GREEN_BG="\033[42m"
+BLACK_BG="\033[40m"
+NO_COLOR="\033[0m"
+PYLINT_FAIL="pylint...................................................................${RED_BG}Failed${NO_COLOR}\n"
+PYLINT_SUCCESS="pylint...................................................................${GREEN_BG}Passed${NO_COLOR}\n"
+PYLINT_SKIPPED="................................................(no files to check)${BLACK_BG}Skipped${NO_COLOR}\n"
+
+MODIFIED_PYTHON_FILES=$(git diff-index --cached --name-only HEAD -- '*.py')
+
+if [ "$MODIFIED_PYTHON_FILES" ]; then
+    PYLINT_OUT=$(./venv/bin/pylint --score=n $MODIFIED_PYTHON_FILES)
```

Comment:
Can this custom hook be replaced by adding pylint to the pre-commit-config.yaml? This way, we would have one less file and dependency to manage (directly, at least) :grimacing:  

```
# pre-commit-config.yaml
...
  - repo: https://github.com/pre-commit/mirrors-pylint
    rev: v2.6.0
    hooks:
    -   id: pylint
```

Reply:
No, it can't. 

"pre-commit runs pylint from an isolated virtualenv. Many of pylint's checks perform dynamic analysis which will fail there. You may find configuring pylint as a local hook more useful."

link: https://github.com/pre-commit/mirrors-pylint

Reply:
I may be wrong, but I think that there is a way to properly configure pre-commit to use pylint as local hook. I remember trying and failing, maybe you could take a look at this?

Reply:
True, I always forget about the `pylint` x `venv` detail :cry: 
I'll dig a little deeper into this today :smile:

Reply:
@lcc  this seems to work with `venv` and is [proper solution](https://pre-commit.com/#repository-local-hooks) and doesn't need and extra directory/file in our repo.

In order to test this, I uninstalled `pylint` systemwide, pruned and bootstrapped the repo.

```
# .pre-commit-config.yaml
...

-   repo: local
    hooks:
    -   id: pylint
        name: pylint
        entry: pylint
        language: system
        types: [python]
```

```
# lala.py

"""Flask App Module"""

from flask import Flask
import json

app = Flask(__name__)

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=5000)
```

```
# output

Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...........................................(no files to check)Skipped
Check for added large files..............................................Passed
black....................................................................Passed
pylint...................................................................Failed
- hook id: pylint
- exit code: 20

************* Module lala
lala.py:4:0: W0611: Unused import json (unused-import)
lala.py:4:0: C0411: standard import "import json" should be placed before "from flask import Flask" (wrong-import-order)

-------------------------------------------------------------------
Your code has been rated at 6.00/10 (previous run: 10.00/10, -4.00)
```

Reply:
Why can't I resolve this comment? :rofl: 
@agms-cesar I believe this may have something to with the [repo permissions](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request). 

>_"You can resolve a conversation in a pull request if you opened the pull request or if you have write access to the repository where the pull request was opened."_

Would you mind adding write permissions to the KNoT team, for this repo? @netoax could help you with that :smile: